src/js/components/Select/__tests__/Select-test.js

```diff
@@ -356,4 +357,25 @@ describe('Select', () => {
 
     expect(document.getElementById('test-select__drop')).not.toBeNull();
   });
+
+  test('renders without icon', () => {
+    const component = renderer.create(
+      <Select id="test-select" options={['one', 'two']} icon={false} />,
+    );
+    expect(component.toJSON()).toMatchSnapshot();
+  });
+
+  test('renders custom icon', () => {
+    const component = renderer.create(
+      <Select id="test-select" options={['one', 'two']} icon={CaretDown} />,
```

Comment:
PropTypes is not liking this one 
```
    ✓ renders custom icon (12ms)

 › 1 snapshot obsolete.
   • Select renders default icon 1
  console.error node_modules/react/node_modules/prop-types/checkPropTypes.js:19
    Warning: Failed prop type: Invalid prop `icon` supplied to `Select`.
        in Select (created by Select)
        in Select (created by Context.Consumer)
        in WithTheme(Select)
```

Reply:
So i had a look and since we are passing the component as `icon={CarretDown}` then this is not a react node but a function. i wonder though what would happen when someone passes an icon which is a class component.

Anyway proposed solution is to use `PropTypes.func` and also change the TS definition. This problem is also present in the Menu component.

Reply:
let me change it.